### PR TITLE
Remove inplace inversion

### PR DESCRIPTION
**Pull request by AlbertMitjans**
- Use `qml.adjoint` instead of `.inv()`
- Remove checks for `QubitStateVector` (they were recently added inside its `__init__` method).

### CODE REVIEW

**Comment by AlbertMitjans** [comment]
[sc-33453]

File: pennylane_qulacs/qulacs_device.py [hunk 1]
```diff
@@ -200,10 +202,6 @@ def _apply_qubit_state_vector(self, op):
         wires = self.map_wires(op.wires)
         input_state = op.parameters[0]
 
-        if len(input_state) != 2 ** len(wires):
```
Comment (rmoyard) [inline pennylane_qulacs/qulacs_device.py:203]: Why were those checks removed?

**Review by rmoyard — APPROVED** [review]
It looks good to me :100: just one question

File: pennylane_qulacs/qulacs_device.py (hunk 1 above)
Comment (AlbertMitjans) [reply pennylane_qulacs/qulacs_device.py:203]: Check the PR description, they were recently added in `QubitStateVector.__init__`

**Pull request merged**